src/main/java/com/zufar/onlinestore/payment/endpoint/PaymentEndpoint.java

```diff
@@ -0,0 +1,80 @@
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+
+import java.util.Objects;
+
+@Slf4j
+@Validated
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(PaymentEndpoint.PAYMENT_URL)
+public class PaymentEndpoint {
+
+    public static final String PAYMENT_URL = "/api/v1/payment";
+
+    private final PaymentApi paymentApi;
+
+    @PostMapping
+    public ResponseEntity<PaymentDetailsWithTokenDto> createPayment(@RequestBody @Valid final CreatePaymentDto createPaymentDto) throws StripeException {
+        PaymentDetailsWithTokenDto createdPayment = paymentApi.createPayment(createPaymentDto);
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .body(createdPayment);
+    }
+
+    @GetMapping("/{paymentId}")
+    public ResponseEntity<PaymentDetailsDto> getPaymentDetails(@PathVariable @Valid @NotNull Long paymentId) {
+        PaymentDetailsDto retrievedPayment = paymentApi.getPaymentDetails(paymentId);
+        if (Objects.isNull(retrievedPayment)) {
+            log.info("Get payment details: not found payment details by id: {}.", paymentId);
+            return ResponseEntity.notFound()
+                    .build();
+        }
+        log.info("Get payment details: payment details: {} successfully retrieved.", retrievedPayment);
+
+        return ResponseEntity.ok()
+                .body(retrievedPayment);
+    }
+
+    /**
+     * This endpoint is used only until we have an implementation of this logic on the frontend side.
+     * It will come in handy for testing the API.
+     */
+    @PostMapping("/method")
+    public ResponseEntity<String> getPaymentMethod(@RequestBody @Valid final CreatePaymentMethodDto createPaymentMethodDto) throws StripeException {
+        String paymentMethodId = paymentApi.createPaymentMethod(createPaymentMethodDto);
+        return ResponseEntity.ok()
+                .body(paymentMethodId);
+    }
+
+    @PostMapping("/event")
+    public ResponseEntity<Void> paymentEventsProcess(
+            @RequestBody @Valid @NotEmpty String paymentIntentPayload,
+            @RequestHeader("Stripe-Signature") @Valid @NotEmpty String stripeSignatureHeader) throws SignatureVerificationException {
+        if (Objects.isNull(paymentIntentPayload) || Objects.isNull(stripeSignatureHeader)) {
```

Comment:
constrained with @NotEmpty must be not null, and its size/length must be greater than zero.

I dont see so much sense to put additional if checks for it

Reply:
Add @NotNull constraint, but NotEmpty annotation validate that string must be not null, and its size/length must be greater than zero, adding @Size doesnt make sense